test/distributed/_composable/fully_shard/test_fully_shard_model_checkpoint.py

```diff
@@ -0,0 +1,188 @@
+        return 2
+
+    @skip_if_lt_x_gpu(2)
+    def test_state_dict_save_load_root_fully_shard(self):
+        """
+        Tests that the full state dict saved from a module with ``fully_shard``
+        applied to the global root matches that of an equivalent local module. Also
+        ensure that this state_dict can be reloaded into a composable module and
+        is equivalent to the original composable module.
+        """
+        local_model = CompositeParamModel(device=torch.device("cuda"))
+        save_composable = copy.deepcopy(local_model)
+        fully_shard(save_composable, policy=ModuleWrapPolicy({UnitModule}))
+        local_sd = local_model.state_dict()
+        composable_sd = save_composable.state_dict()
+        self._check_state_dict_parity(local_sd, composable_sd)
+
+        # Validate load
+        load_composable = fully_shard(
+            copy.deepcopy(local_model), policy=ModuleWrapPolicy({UnitModule})
+        )
+        _zero_model(load_composable, summon_full=False)
+        for p in load_composable.parameters():
+            self.assertEqual(p.sum(), 0)
+
+        sd = {k: v.clone() for k, v in composable_sd.items()}
+        load_composable.load_state_dict(sd)
+        self._check_model_parity(load_composable, save_composable)
+
+    @skip_if_lt_x_gpu(2)
+    def test_state_dict_save_load_submodule_fully_shard(self):
+        """
+        Tests that the full state dict saved from a module with ``fully_shard``
+        applied on submodules matches that of an equivalent local module. Also
+        ensures that this state_dict can be reloaded into a composable module and
+        is equivalent to the original composable module.
+        """
+        local_model = CompositeParamModel(device=torch.device("cuda"))
+
+        def _create_fully_shard_on_submodules(mod: nn.Module):
+            fully_shard(mod.u1)
+            fully_shard(mod.u2)
+            return mod
+
+        save_composable = copy.deepcopy(local_model)
+        save_composable = _create_fully_shard_on_submodules(save_composable)
+        local_sd = local_model.state_dict()
+        composable_sd = save_composable.state_dict()
+        self._check_state_dict_parity(local_sd, composable_sd)
+
+        # Validate load
+        load_composable = copy.deepcopy(local_model)
+        load_composable = _create_fully_shard_on_submodules(load_composable)
+        _zero_model(load_composable, summon_full=False)
+        for p in load_composable.parameters():
+            self.assertEqual(0, p.sum())
+
+        sd = {k: v.clone() for k, v in composable_sd.items()}
+        load_composable.load_state_dict(sd)
+        self._check_model_parity(load_composable, save_composable)
+
+    @skip_if_lt_x_gpu(2)
+    def test_state_dict_save_load_flow(self):
+        """
+        E2E test of save + load with rank0_only + CPU offload for TransformerWithSharedParams
+        on the composable path.
+        """
+        self.run_subtests(
```

Comment:
This is the only real change, where I knocked out a to-do to use `run_subtests`.